tutor/commands/upgrade/compose.py

```diff
@@ -158,6 +158,36 @@ def upgrade_from_olive(context: click.Context, config: Config) -> None:
     upgrade_mongodb(context, config, "4.2.17", "4.2")
     upgrade_mongodb(context, config, "4.4.22", "4.4")
 
+    upgrade_to_redwood_onwards, new_mysql_docker_image = (
```

Comment:
This part is just too confusing. I suggest the following:

```
intermediate_mysql_docker_image = get_intermediate_mysql_upgrade(config)
if not intermediate_mysql_docker_image:
    return

click.echo(fmt.title(f"Upgrading MySQL to {intermediate_mysql_docker_image}"))
mysql_docker_image = config["DOCKER_IMAGE_MYSQL"]
config["DOCKER_IMAGE_MYSQL"] = intermediate_mysql_docker_image
tutor_env.save(context.obj.root, config)
context.invoke(do.commands["init"], limit="mysql")
```

Note that the mysql init task does the exact same command as `is_mysql_service_ready`, which means that we can get rid of this function.

Reply:
Huh, I didn't know we could call init commands directly in the tutor codebase. Let me try this out.

Reply:
This worked out perfectly, thanks for the heads up.